boards/cargo/tock_flags.toml

```
@@ -0,0 +1,37 @@
+# Licensed under the Apache License, Version 2.0 or the MIT License.
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+# Copyright Tock Contributors 2024.
+
+[build]
+rustflags = [
+  # Tell rustc to use the LLVM linker. This avoids needing GCC as a dependency
+  # to build the kernel.
+  "-C", "linker=rust-lld",
+  # Use the LLVM lld executable with the `-flavor gnu` flag.
+  "-C", "linker-flavor=ld.lld",
+  # Use static relocation model. See https://github.com/tock/tock/pull/2853
+  "-C", "relocation-model=static",
+  # lld by default uses a default page size to align program sections. Tock
+  # expects that program sections are set back-to-back. `-nmagic` instructs the
+  # linker to not page-align sections.
+  "-C", "link-arg=-nmagic",
+  # Identical Code Folding (ICF) set to all. This tells the linker to be more
+  # aggressive about removing duplicate code. The default is `safe`, and the
+  # downside to `all` is that different functions in the code can end up with
+  # the same address in the binary. However, it can save a fair bit of code
+  # size.
+  "-C", "link-arg=-icf=all",
+  # Opt-in to Rust v0 symbol mangling scheme.  See
+  # https://github.com/rust-lang/rust/issues/60705 and
+  # https://github.com/tock/tock/issues/3529.
+  # https://github.com/rust-lang/rust/pull/89917 tracks this becoming the default, then we can remove this.
+  "-C", "symbol-mangling-version=v0",
+  # Enable link-time-optimization
+  "-C", "lto",
+]
+
+[target.test]
+rustdocflags = [
+  "-D", "warnings",
+  "--document-private-items"
+]
```

```suggestion
rustdocflags = [
  "-D", "warnings",
  "--document-private-items"
]
```

Why did you set this for target.test and not build?

---

Excellent question!